test-data/unit/check-newsemanal.test

```diff
@@ -339,14 +339,25 @@ def main() -> None:
             x  # E: Name 'x' is not defined
 
 [case testNewAnalyzerCyclicDefinitions]
-gx = gy  # E: Cannot determine type of 'gy'
+gx = gy  # E: Cannot resolve name "gy", possible cyclic definition
 gy = gx
 def main() -> None:
     class C:
         def meth(self) -> None:
-            lx = ly  # E: Cannot determine type of 'ly'
+            lx = ly  # E: Cannot resolve name "ly", possible cyclic definition
             ly = lx
 
+[case testNewAnalyzerCyclicDefinitionCrossModule]
+import b
+[file a.py]
+import b
+x = b.x  # E: Cannot determine type of 'x'
+[file b.py]
+import a
+x = a.x  # E: Cannot resolve attribute "x", possible cyclic definition \
+         # E: Module has no attribute "x"
```

Comment:
Would it be easy to get rid of the second error, as it seems redundant and a bit confusing.

Reply:
Opened https://github.com/python/mypy/issues/6551 for this.